### PR TITLE
Computer smoothing from Bay

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/ss220-space/Baystation12/pull/350 (hope you like speaking russian, kiddo)

## About The Pull Request
Ports computer smoothing from a bay downstream

Adds new smoothing group, to work with directional objects, as smoothing is traditionally only has been used for walls and structures. Not stuff with faces and dirs

## Why It's Good For The Game
It looks good. It makes administrative centers cluttered with all these damn consoles look semi-organized and sci-fi.

It fits Bridge and shuttle vibes.

A bunch of cluttered individual consoles doesn't look like a control panel, it looks like a mess. 

## Testing Photographs and Procedure

security

hop

beauty

asfds

securityoffice

## Changelog
:cl: RKz, HowToLoLu/Dakae, Baystation Spriters(ImJustKisik)
add: new smoothing group, includes computers and modular consoles
code: new smoothing_flag, directional smoothing, for stuff with dirs and faces
/:cl:

### CODE REVIEW

**Comment by itsmeow** [comment]
I don't like the north and south icons, they look like it's extruding or like a mini console. The ones from the side look nice.

File: code/__DEFINES/icon_smoothing.dm [hunk 1]
```diff
@@ -20,6 +21,7 @@ DEFINE_BITFIELD(smoothing_flags, list(
 	"SMOOTH_BORDER" = SMOOTH_BORDER,
 	"SMOOTH_QUEUED" = SMOOTH_QUEUED,
 	"SMOOTH_OBJ" = SMOOTH_OBJ,
+	"SMOOTH_DIRECTIONALS" = SMOOTH_OBJ,
```
Comment (PowerfulBacon) [inline code/__DEFINES/icon_smoothing.dm]: This has the wrong bit associated with it

File: code/__DEFINES/icon_smoothing.dm [hunk 2]
```diff
@@ -114,6 +116,10 @@ DEFINE_BITFIELD(smoothing_flags, list(
 #define SMOOTH_GROUP_HEDGE_FLUFF S_OBJ(65)				///obj/structure/fluff/hedge
 #define SMOOTH_GROUP_SHUTTLE_PARTS S_OBJ(66)			///obj/structure/window/shuttle, /obj/structure/window/plasma/reinforced/plastitanium, /turf/closed/indestructible/opsglass, /obj/structure/shuttle
 #define SMOOTH_GROUP_CLEANABLE_DIRT	S_OBJ(67)			///obj/effect/decal/cleanable/dirt
+///obj/machinery/computer/_computer
+#define SMOOTH_GROUP_COMPUTERS S_OBJ (68)
+///obj/machinery/modular_computer/console
+#define SMOOTH_GROUP_CONSOLE S_OBJ (69)
```
Comment (PowerfulBacon) [inline code/__DEFINES/icon_smoothing.dm]: Is this ever actually used?

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Kinda wish we had unit tests to ensure that anything being smoothed has the correct icons for smoothing so we don't have problems in the future, but that's non-blocking.

File: code/__DEFINES/icon_smoothing.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/__DEFINES/icon_smoothing.dm]: No, I used it during testing and later deleted it. I'll remove it

Comment (Tsar-Salat) [reply code/__DEFINES/icon_smoothing.dm]: removed 7b090bb

File: code/__DEFINES/icon_smoothing.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/__DEFINES/icon_smoothing.dm]: fixed 7b090bb

**Comment by Tsar-Salat** [comment]
> Kinda wish we had unit tests to ensure that anything being smoothed has the correct icons for smoothing so we don't have problems in the future

Bit above my knowledge atm.

So far, the only smoothing group needing unique behavior is computers, as walls and tables all used the same stuff. Not really worried about the maintainability if its limited to one thing.

Ive spawned in multiple different orientations that players may use, so I think our bases are covered for now

**Review by PowerfulBacon — APPROVED** [review]

**Comment by itsmeow** [comment]
image

**Pull request merged**